sycl/test-e2e/KernelAndProgram/free_function_apis.cpp

```diff
@@ -0,0 +1,227 @@
+// REQUIRES: aspect-usm_shared_allocations
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// The name mangling for free function kernels currently does not work with PTX.
+// UNSUPPORTED: cuda
+
+#include <iostream>
+#include <sycl/detail/core.hpp>
+#include <sycl/ext/oneapi/free_function_queries.hpp>
+#include <sycl/usm.hpp>
+
+using namespace sycl;
+
+SYCL_EXTERNAL
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(
+    (ext::oneapi::experimental::nd_range_kernel<2>))
+void ff_2(int *ptr, int start) {
+  int(&ptr2D)[4][4] = *reinterpret_cast<int(*)[4][4]>(ptr);
+  nd_item<2> Item = ext::oneapi::this_work_item::get_nd_item<2>();
+  id<2> GId = Item.get_global_id();
+  id<2> LId = Item.get_local_id();
+  ptr2D[GId.get(0)][GId.get(1)] = LId.get(0) + LId.get(1) + start;
+}
+
+// Templated free function definition.
+template <typename T>
+SYCL_EXTERNAL SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((
+    ext::oneapi::experimental::single_task_kernel)) void ff_3(T *ptr, T start) {
+  int(&ptr2D)[4][4] = *reinterpret_cast<int(*)[4][4]>(ptr);
+  nd_item<2> Item = ext::oneapi::this_work_item::get_nd_item<2>();
+  id<2> GId = Item.get_global_id();
+  id<2> LId = Item.get_local_id();
+  ptr2D[GId.get(0)][GId.get(1)] = LId.get(0) + LId.get(1) + start;
+}
+
+// Explicit instantiation of free function.
+template void ff_3(int *ptr, int start);
+
+// A plain function that is not a free function.
+void ff_4(int *ptr, int start) {}
+
+bool test_kernel_apis(queue Queue) {
+  bool Pass = true;
+
+#ifndef __SYCL_DEVICE_ONLY__
+  // Check for a free function, which is known to be a free function.
+  // Expect: true.
+  bool Pass1 = ext::oneapi::experimental::is_nd_range_kernel_v<ff_2, 2>;
+  std::cerr << "Pass1=" << Pass1 << std::endl;
```

Comment:
```suggestion
  std::cout << "Pass1=" << Pass1 << std::endl;
```
I don't think `cerr` should be used here since this is printed unconditionally, even if `Pass1` is correct. The same goes for the rest of the file.

Reply:
I've changed the output to use std::cout.